cocos/core/gfx/webgl/webgl-device.ts

```diff
@@ -245,6 +213,166 @@ export class WebGLDevice extends Device {
+        this._formatFeatures.fill(FormatFeatureBit.NONE);
+
+        this._textureExclusive.fill(true);
+
+        const completeFeature: FormatFeature = FormatFeatureBit.RENDER_TARGET | FormatFeatureBit.SAMPLED_TEXTURE
+            | FormatFeatureBit.LINEAR_FILTER;
+        let tempFeature: FormatFeature = completeFeature;
+
+        this._formatFeatures[Format.RGB8] = tempFeature;
+        this._formatFeatures[Format.R5G6B5] = tempFeature;
+        this._textureExclusive[Format.R5G6B5] = false;
+
+        this._formatFeatures[Format.RGBA8] = tempFeature;
+        this._formatFeatures[Format.RGBA4] = tempFeature;
+        this._textureExclusive[Format.RGBA4] = false;
+
+        this._formatFeatures[Format.RGB5A1] = tempFeature;
+        this._textureExclusive[Format.RGB5A1] = false;
+
+        this._textureExclusive[Format.DEPTH] = false;
+        this._textureExclusive[Format.DEPTH_STENCIL] = false;
+
+        this._formatFeatures[Format.R8I] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RG8I] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RGB8I] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RGBA8I] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+
+        this._formatFeatures[Format.R8UI] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RG8UI] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RGB8UI] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RGBA8UI] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+
+        this._formatFeatures[Format.R8I] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RG8I] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RGB8I] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RGBA8I] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+
+        this._formatFeatures[Format.R8UI] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RG8UI] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RGB8UI] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RGBA8UI] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+
+        this._formatFeatures[Format.R32F] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RG32F] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RGB32F] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+        this._formatFeatures[Format.RGBA32F] |= FormatFeatureBit.VERTEX_ATTRIBUTE;
+
+        if (exts.EXT_sRGB) {
+            tempFeature = FormatFeatureBit.RENDER_TARGET | FormatFeatureBit.SAMPLED_TEXTURE
+                | FormatFeatureBit.LINEAR_FILTER;
+            this._formatFeatures[Format.SRGB8] = tempFeature;
+            this._formatFeatures[Format.SRGB8_A8] = tempFeature;
+
+            this._textureExclusive[Format.SRGB8_A8] = false;
+        }
+
+        if (exts.WEBGL_depth_texture) {
```

Comment:
if this extension is not present, then these two format won't be able to be sampled from (renderbuffer only)